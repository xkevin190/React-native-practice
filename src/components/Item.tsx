import * as React from 'react';
import {Text, View, StyleSheet, TouchableOpacity} from 'react-native';
import {item} from '../state/aplication/type';
import Icon from 'react-native-vector-icons/Ionicons';
import {colors} from '../constants/colors';
import {normalize} from '../constants/utils';
interface ItemProps {
  gender: string;
  name: string;
  birth_year: string;
  homeworld: string;
  favorite: string;
  action: (name: string) => void;
}

const Item = (props: ItemProps) => {
  return (
    <View style={styles.container}>
      <View style={styles.containerFavoriteIcon}>
        <TouchableOpacity onPress={() => props.action(props.name)}>
          <Icon
            size={normalize(15)}
            color={colors.primary}
            name={props.favorite ? 'heart' : 'heart-outline'}
          />
        </TouchableOpacity>
      </View>
      <Text style={styles.texTitle}>{props.name}</Text>
      <Text style={styles.textSpam}>
        {`${props.gender}  |  ${props.birth_year}`}
      </Text>
      <View style={styles.locationContainer}>
        <Icon
          color={colors.primary}
          size={normalize(13)}
          name="location-outline"
        />
        <Text style={[styles.texTitle, styles.locationText]}>
          {props.homeworld}
        </Text>
      </View>
    </View>
  );
};

export default React.memo(Item);

const styles = StyleSheet.create({
  container: {
    marginTop: 20,
    marginHorizontal: 20,
    borderBottomWidth: 1,
    paddingBottom: 20,
    borderBottomColor: colors.separator,
  },
  texTitle: {
    fontFamily: 'Urbanist',
    color: colors.primary,
    fontSize: normalize(13),
    fontWeight: '600',
  },

  containerFavoriteIcon: {
    alignItems: 'flex-end',
    bottom: -10,
  },

  textSpam: {
    paddingTop: 10,
    fontFamily: 'Urbanist',
    color: colors.textSpam,
    fontSize: normalize(12),
    fontWeight: '400',
  },
  locationContainer: {
    flexDirection: 'row',
    marginTop: 10,
    backgroundColor: colors.secondary,
    width: '30%',
    paddingVertical: 3,
    paddingHorizontal: 8,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 6,
  },

  locationText: {
    fontSize: normalize(12),
    paddingLeft: 3,
  },
});
