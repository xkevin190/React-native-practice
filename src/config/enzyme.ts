import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

/**
 * @tech-debt
 *
 * Update enzyme to support react 18
 */
Enzyme.configure({ adapter: new Adapter() });
