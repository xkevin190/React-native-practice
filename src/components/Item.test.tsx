import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import Item from './Item';

describe('Item (Unit-Test)', () => {
  let wrapper: ShallowWrapper;

  const favoriteAction = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Item
        gender={'test'}
        name={'test'}
        birth_year={'test'}
        homeworld={'test'}
        favorite={false}
        action={favoriteAction}
      />,
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('the favorite icon should be disabled', () => {
    expect(wrapper.find('Icon').first().prop('name')).toBe('heart-outline');
  });

  it('the favorite icon should be actived', () => {
    wrapper = shallow(
      <Item
        gender={'test'}
        name={'test'}
        birth_year={'test'}
        homeworld={'test'}
        favorite={true}
        action={favoriteAction}
      />,
    );
    expect(wrapper.find('Icon').first().prop('name')).toBe('heart');
  });

  it('closeDrawer function should executed', () => {
    wrapper.find('TouchableOpacity').first().simulate('press');
    expect(favoriteAction).toBeCalledWith('test');
  });
});
