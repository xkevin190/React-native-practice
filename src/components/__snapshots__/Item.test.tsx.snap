// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item (Unit-Test) should match snapshot 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#463B5E",
      "borderBottomWidth": 1,
      "marginHorizontal": 20,
      "marginTop": 20,
      "paddingBottom": 20,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "Urbanist",
          "fontSize": 29,
          "fontWeight": "600",
        }
      }
    >
      test
    </Text>
    <TouchableOpacity
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "width": 40,
        }
      }
    >
      <Icon
        allowFontScaling={false}
        color="#ffffff"
        name="heart-outline"
        size={33}
      />
    </TouchableOpacity>
  </View>
  <Text
    style={
      Object {
        "color": "rgba(255, 255, 255, 0.7)",
        "fontFamily": "Urbanist",
        "fontSize": 26,
        "fontWeight": "400",
        "paddingTop": 10,
      }
    }
  >
    test  |  test
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2f2a35",
        "borderRadius": 6,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 10,
        "paddingHorizontal": 8,
        "paddingVertical": 3,
        "width": "30%",
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="#ffffff"
      name="location-outline"
      size={29}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Urbanist",
            "fontSize": 29,
            "fontWeight": "600",
          },
          Object {
            "fontSize": 26,
            "paddingLeft": 3,
          },
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;
